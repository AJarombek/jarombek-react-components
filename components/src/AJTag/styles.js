/**
 * Stylesheet for the AJTag component using JSS.
 * Author: Andrew Jarombek
 * Date: 5/17/2020
 */

export default {
  ajTag: ({ backgroundColor }) => ({
    backgroundColor,
    display: 'inline-block',
    cursor: 'pointer',
    padding: '0 10px',
    borderRadius: '3px',
    boxShadow: '0 1px 3px rgba(170, 170, 170, 0.1)',
    transition: 'all .4s ease',

    '&:hover': {
      boxShadow: '0 1px 3px rgba(170, 170, 170, 0.5)',
    },
  }),
};
