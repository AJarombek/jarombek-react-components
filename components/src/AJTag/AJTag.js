/**
 * Tag component which displays a node.
 * @author Andrew Jarombek
 * @since 5/17/2020
 */

import React from 'react';
import PropTypes from 'prop-types';
import {createUseStyles} from 'react-jss';
import classnames from 'classnames';

import styles from './styles';

const useStyles = createUseStyles(styles);

const AJTag = ({ content, color }) => {
  const backgroundColor = color ?? 'rgba(204, 204, 204, 0.6)';
  const classes = useStyles({ backgroundColor });

  return (
    <div className={classnames(classes.ajTag, 'aj-tag')}>
      {content}
    </div>
  );
};

AJTag.propTypes = {
  content: PropTypes.node.isRequired,
  color: PropTypes.string
};

export default AJTag;
