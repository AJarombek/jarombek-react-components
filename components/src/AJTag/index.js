import AJTag from './AJTag';

export default AJTag;
