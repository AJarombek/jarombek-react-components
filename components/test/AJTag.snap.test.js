/**
 * Snapshot test for the {@link ../../components/AJTag} component.
 * @author Andrew Jarombek
 * @since 5/17/2020
 */

import React from 'react';
import AJTag from '../src/AJTag/AJTag';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<AJTag content="Snapshot Tag"/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with all props', () => {
  const tree = renderer.create(
    <AJTag content={<p>Snapshot Tag</p>} color="#900"/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
