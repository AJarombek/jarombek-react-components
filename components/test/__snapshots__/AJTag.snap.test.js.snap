// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ajTag-0-2-1 ajTag-d0-0-2-2 aj-tag"
>
  Snapshot Tag
</div>
`;

exports[`renders correctly with all props 1`] = `
<div
  className="ajTag-0-2-1 ajTag-d1-0-2-3 aj-tag"
>
  <p>
    Snapshot Tag
  </p>
</div>
`;
